testing.md

```diff
@@ -0,0 +1,83 @@
+# Testing
+
+## Recommended Packages
+
+## Mocha - https://mochajs.org
+
+[Mocha][] is the most _depended-upon_ package on the npm registry. It is a mature (created in 2011) and stable project. While Node.js is the main area of focus and support, Mocha can also be used in a browser context.  Mocha values flexibility, and as such, is considered an _unopinionated_ framework.  Mocha has a large ecosystem of plugins and extensions.
+
+Mocha is an [OpenJS Foundation][] project with open governance.  An IBMer (@boneskull) is a lead maintainer of the project.
+
+## Jest - https://jestjs.io
+
+[Jest][] is a popular testing framework from Facebook.  It excels at testing React and other component-based web applications, and can be used in other contexts.  It is considered an _opinionated_ framework, as it provides its own set of assertions, spies/stubs/mocks, and other features (such as snapshot testing and code coverage) out-of-the-box.
+
+Jest is owned by Facebook.
+
+## Guidance
```

Comment:
for me, based on meeting, add additional guidance on jest

Reply:
will you provide a "suggestion" here in the review?

Reply:
yep it is in my todo list! hopefully can get to it before next week's meeting.

Reply:
np.  I cannot make the meetings due to my schedule, but happy to discuss any of this here or otherwise!

Reply:
@boneskull, I think you mentioned a while back that you had collected a list of the IBM projects using Mocha, can you post links to that data here?

Reply:
or if its IBM internal info, then send to the team in some other way.

Reply:
heres new section i put together

```
#### Using Jest Snapshots

Jest provides some unique features compared to Mocha. One such feature is the use of [snapshots][]. Snapshots provide an easy to use way of testing output of a function and saving the result as a snapshot artifact, which can then be used to compare against as a test.

```
import React from 'react';
import renderer from 'react-test-renderer';
import Link from '../Link.react';

it('renders correctly', () => {
  const tree = renderer
    .create(<Link page="http://www.facebook.com">Facebook</Link>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
```

In the case that snapshot tests do not match the artifact, it will report a test failure. The test can be easily fixed by running `jest test --updateSnapshot`, however it is advised you only do so until after validating/verifying the results of the new snapshot artifact.
```

Reply:
Amazing suggestion @roastlechon +1 to add it